feat(Alert): add size props; add css-api; refactor component

benax-se opened the pull request.

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -3,12 +3,129 @@
+        color: var(--g-alert-title-color, var(--g-color-text-primary));
+        font-size: var(--g-alert-title-font-size, var(--_--title-font-size));
+        font-weight: var(--g-alert-title-font-weight, var(--_--title-font-weight));
+        line-height: var(--g-alert-title-line-height, var(--_--title-line-height));
```
There is no need of font weight CSS vars, user can do the following:
```jsx
<Alert title={<MyBoldTitle/>} />
```

Let's keep only `font-size` and `line-height`

benax-se replied:
fixed

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -3,12 +3,129 @@
+    }
+
+    &__title {
+        color: var(--g-alert-title-color, var(--g-color-text-primary));
```
```suggestion
        color: var(--g-alert-title-text-color, var(--g-color-text-primary));
```

benax-se replied:
fixed

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -3,12 +3,129 @@
+    }
+
+    &__message {
+        color: var(--g-alert-message-color, var(--g-color-text-complementary));
```
```suggestion
        color: var(--g-alert-message-text-color, var(--g-color-text-complementary));
```

benax-se replied:
fixed

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -17,5 +134,6 @@ $block: '.#{variables.$ns}alert';
 
     &__close-btn {
         flex-shrink: 0;
+        margin-inline-start: var(--g-alert-close-btn-indent, var(--_--close-btn-indent));
```
What's the purpose of manipulating both margin and relative position? There are already usage of `--_--close-btn-offset` above.

benax-se replied:
after discussion with designer we decided to remove additional offset

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -3,12 +3,129 @@
+    &__main {
+        flex-grow: 1;
+        display: flex;
+        gap: var(--g-alert-content-gap, var(--_--content-gap));
```
We should specify in CSS API only the parts we publically declare. The content name is the internal structure. This should be variables like: `g-alert-message-margin`, `g-alert-actions-margin` and so on

benax-se replied:
fixed

amje commented on src/components/Alert/Alert.tsx:
```diff
@@ -13,69 +12,82 @@ import {AlertIcon} from './AlertIcon';
         view = 'filled',
+        size = 'm',
         layout = 'vertical',
+        closeBtnSize = 'm',
```
```suggestion
        closeButttonSize = 'm',
```

benax-se replied:
fixed

amje commented on src/components/Alert/Alert.scss:
```diff
@@ -3,12 +3,129 @@
+    }
+
+    &__icon {
+        vertical-align: text-bottom;
```
Icon is now not aligned properly with the title. Icon should be a bit under the text baseline

benax-se replied:
fixed

amje commented:
@IsaevAlexandr Take a look please

benax-se commented on src/components/Alert/Alert.scss:
```diff
@@ -3,19 +3,140 @@
+
+        &_align_baseline {
+            justify-content: start;
+        }
```
fixed

benax-se commented:
@amje i've fixed all issues

There are difference in screenshots because stable version has a bug - 'baseline' align does not applied as default (as it's mentioned in the documentation), in the new version i've fixed this.

amje commented on package.json:
```diff
@@ -122,7 +122,7 @@
     "playwright:clear-cache": "rm -rf ./playwright/.cache",
     "playwright:docker": "./playwright/run-docker-command.sh npm run playwright --",
-    "playwright:docker:update": "./playwright/run-docker-command.sh npm run playwright:update --",
+    "playwright:docker:update": "./playwright/run-docker-command.sh npm run playwright:update src/components/Alert/__tests__/Alert.visual.test.tsx --",
```
dev code

benax-se replied:
thx, removed this

benax-se committed:
feat(Alert): add size props; add css-api; refactor component
